ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/FileWatcher.java

```diff
@@ -95,23 +96,23 @@ public void run() {
   }
 
   @SuppressWarnings("unchecked")
-  @SuppressFBWarnings(
-      value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
-      justification = "Null check on file.getParent() is present above"
-  )
   private void handleNextWatchNotification() throws InterruptedException {
-    // wait for key to be signalled
-    final WatchKey key = watchService.take();
-    log.info("Watch Key notified");
+    final WatchKey key = watchService.poll(1, TimeUnit.SECONDS);
+    if (key == null) {
```

Comment:
should we backoff or something here? I feel like polling every second is going to make this thread take up way more resources that it needs to (how often do certs change? monthly?)

I'm also not entirely sure why this poll is necessary (why not just use `take`)?

Reply:
+1 I'm confused why this change was made.

Reply:
`take` blocks and requires the use of interrupts to shutdown the thread.  This is generally considered an anti-pattern and goes against the "use exceptions for exceptional cases only" paradigm. 

Exiting the poll every second,,and then immediately looping and re-entering the poll, will take almost no CPU cycles and allows it to exit gracefully.

Reply:
> This is generally considered an anti-pattern and goes against the "use exceptions for exceptional cases only" paradigm.

I agree that requiring interrupts to shutdown the thread should be avoided, but I've seen different workarounds to this problem (specifically when using `BlockingQueue`) - the most popular of which is adding a sentinel into the queue to wake up the blocked thread.

I'm not familiar with `WatchService`, but is it possible to do something similar (e.g. touch a lock file)? I understand your reasoning now, so if there isn't clean we can do then I'm happy proceeding with this

Reply:
Sounds like more code for the sake of it.   Are you really worried about a thread waking every second and looping?